### PR TITLE
Improve env vars explanations in agent and backend references

### LINKED ISSUE #2257: Update Sensu Go agent environment variable file documentation to match backend more closely
### Description
The current section desribing how to use environment variables on disk with the Sensu Go agent is missing some critical details on requirements and behavior, which are present in the backend.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [X] improvement (improve docs content or format)

### Affected pages

https://docs.sensu.io/sensu-go/latest/reference/agent/#use-environment-variables-with-the-sensu-agent

Has some talk about how to get environment variables enabled, but the backend one has a lot more details on requirements and behavior.

### Expected behavior
Provide as much information as possible!

### Current behavior
Some important details are missing when using the agent to configure environment variables.

### Corrections needed
Ask @rgeniesse for them or I'll just take it.

### Screenshots
None.

### Additional context

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Clarify and reorganize environment variables sections in agent reference
- Add Use env vars section in backend reference
- Update TOC in agent reference to include General config flags heading

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2257

### CODE REVIEW

File: content/sensu-go/5.18/reference/agent.md [hunk 1]
```diff
@@ -1437,9 +1438,11 @@ $ sudo systemctl restart sensu-agent
 
 {{< /language-toggle >}}
 
-## Use environment variables with the Sensu agent
+### Use environment variables with the Sensu agent
 
-After you [configure][50] your sensu-agent service to read environment variables from `/etc/default/sensu-agent` (Debian/Ubuntu) or `/etc/sysconfig/sensu-agent` (RHEL), any environment variables you provide there will be available to check and hook commands executed by the Sensu agent.
+After you [configure][50] your sensu-agent service to read environment variables from `/etc/default/sensu-agent` (Debian/Ubuntu) or `/etc/sysconfig/sensu-agent` (RHEL), you can create custom environment variables there (in addition the environment variables Sensu already recognizes, such as those listed in the [agent configuration tables][16]).
```
Comment (rgeniesse) [inline content/sensu-go/5.18/reference/agent.md]: This reads a bit odd to me, but maybe I am just not understanding it all the way. Actually even the previous language was a bit confusing.

I read this as "after you configure your sensu-agent service to read environment variables....you can then create environment variables there." Where is there? Is this saying one thing, then saying the same thing again?

Comment (hillaryfraley) [reply content/sensu-go/5.18/reference/agent.md]: Yes, I see what you're saying, thank you. Let me try to clarify...

**Comment by hillaryfraley** [comment]
@rgeniesse I reorganized into steps -- does this make more sense to you? It seems clearer to me, but would like to get your feedback.

File: content/sensu-go/5.18/reference/agent.md [hunk 2]
```diff
@@ -1402,47 +1403,75 @@ example               | {{< highlight shell >}}"enable_env": true{{< /highlight
...
 
 {{< highlight "RHEL/CentOS" >}}
 $ sudo touch /etc/sysconfig/sensu-agent
 {{< /highlight >}}
+     
+     {{< /language-toggle >}}
 
-{{< /language-toggle >}}
+2. Make sure the environment variable is named correctly.
+All environment variable names must begin with `SENSU_`.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/agent.md]:
```suggestion
All environment variables controlling Sensu configuration begin with `SENSU_`.
```

File: content/sensu-go/5.18/reference/backend.md [hunk 3]
```diff
@@ -1128,28 +1130,38 @@ etcd-quota-backend-bytes: 4294967296{{< /highlight >}}
...
+    
 {{< highlight "RHEL/CentOS" >}}
 $ sudo touch /etc/sysconfig/sensu-backend
 {{< /highlight >}}
+     
+     {{< /language-toggle >}}
 
-{{< /language-toggle >}}
+2. Make sure the environment variable is named correctly.
+All environment variable names must begin with `SENSU_`.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/backend.md]:
```suggestion
All environment variables controlling Sensu configuration begin with `SENSU_`.
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.18/reference/agent.md [hunk 4]
```diff
@@ -1402,47 +1403,75 @@ example               | {{< highlight shell >}}"enable_env": true{{< /highlight
...
+{{< highlight "RHEL/CentOS" >}}
+$ sudo systemctl restart sensu-agent
+{{< /highlight >}}
 
-## Use environment variables with the Sensu agent
+     {{< /language-toggle >}}
 
-After you [configure][50] your sensu-agent service to read environment variables from `/etc/default/sensu-agent` (Debian/Ubuntu) or `/etc/sysconfig/sensu-agent` (RHEL), any environment variables you provide there will be available to check and hook commands executed by the Sensu agent.
+{{% notice note %}}
+**NOTE**: Sensu includes an environment variable for each backend configuration flag.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/agent.md]:
```suggestion
**NOTE**: Sensu includes an environment variable for each agent configuration flag.
```

File: content/sensu-go/5.18/reference/backend.md [hunk 5]
```diff
@@ -1128,28 +1130,38 @@ etcd-quota-backend-bytes: 4294967296{{< /highlight >}}
...
-{{< language-toggle >}}
+1. Create the files from which the `sensu-backend` service configured by our supported packages will read environment variables: `/etc/default/sensu-backend` for Debian/Ubuntu systems or `/etc/sysconfig/sensu-backend` for RHEL/CentOS systems.
 
+     {{< language-toggle >}}
+     
 {{< highlight "Ubuntu/Debian" >}}
 $ sudo touch /etc/default/sensu-backend
 {{< /highlight >}}
-
+
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/backend.md]:
```suggestion
```

**Review by cwjohnston — APPROVED** [review]
Looks good aside from a couple suggestions here. Thanks! 👍

File: content/sensu-go/5.18/reference/backend.md [hunk 6]
```diff
@@ -1128,28 +1130,37 @@ etcd-quota-backend-bytes: 4294967296{{< /highlight >}}
...
-{{< language-toggle >}}
+1. Create the files from which the `sensu-backend` service configured by our supported packages will read environment variables: `/etc/default/sensu-backend` for Debian/Ubuntu systems or `/etc/sysconfig/sensu-backend` for RHEL/CentOS systems.
 
+     {{< language-toggle >}}
+     
 {{< highlight "Ubuntu/Debian" >}}
 $ sudo touch /etc/default/sensu-backend
 {{< /highlight >}}
-
 {{< highlight "RHEL/CentOS" >}}
```
Comment (hillaryfraley) [inline content/sensu-go/5.18/reference/backend.md:1147]:
```suggestion

{{< highlight "RHEL/CentOS" >}}
```

**Pull request merged**